src/core/tests/mutations.test.tsx

```diff
@@ -56,6 +56,28 @@ describe('mutations', () => {
     expect(result).toBe(result)
   })
 
+  test('executeMutation should propagate error to the mutation cache after retrier rejects', async () => {
+    const key = queryKey()
+
+    const error = new Error("bad error / network issue");
+    queryClient.setMutationDefaults(key, {
+      mutationFn: async () => { throw error},
+      retry: false
+    })
+
+    try{
+      await queryClient.executeMutation({
+        variables: 'todo',
+        mutationKey: key,
+      })
+    }
+    catch (e){ expect(e).toEqual(error);}
+
+    const cache = queryClient.getMutationCache().getAll().find(x => x.state.error);
+
+    expect(cache?.state.error).toEqual(error)
+  })
```

Comment:
I tried this test on `master` and it's also green there. I'm still no sure what exactly the problem is we're trying to solve here, but it's not documented with this test. We'd need a test that _fails_ on `master` and _works_ on this PR please.

Reply:
Sure, I can update it to fail on master.

Here is my reasoning behind why I did this:

The reason nothing fails on master is because nothing should have been broken. This test captures it.

You can see in line [225](https://github.com/tannerlinsley/react-query/pull/3391/files#diff-9c4fdb8c2806ce1f8b1e9a48d82639ac533c39eecc9ef1f1c6cd18562642634bL225) has been removed. This would cause this test to fail.

**Current branch pass**
It passes because onError is implemented on executeMutation which triggers a dispatch when the promise is rejected by the retyer.

**Master branch pass**
It passes in master because master implements line  [225](https://github.com/tannerlinsley/react-query/pull/3391/files#diff-9c4fdb8c2806ce1f8b1e9a48d82639ac533c39eecc9ef1f1c6cd18562642634bL225) which by defaults triggers a dispatch.

**Why do this then?**

This I am not 100% sure of, but triggering a dispatch at the retryer level allows us to hit the error boundary compared to the earlier dispatch that happens before the error is thrown  [225](https://github.com/tannerlinsley/react-query/pull/3391/files#diff-9c4fdb8c2806ce1f8b1e9a48d82639ac533c39eecc9ef1f1c6cd18562642634bL225) .


**Improvement I can make**

If I test the retryer's resolve error gets called, would ensure a fail on master.  I can do that, just don't know how to for now xD.

I will try to re-write the test to fail on master by implementing the above test (see if config.onError gets called)

I hope that makes sense, let me know if there is anything else.

Reply:
if this is about error boundaries, please create a test case that tries to show an error boundary, as that should fail then. We already have tests for error boundaries.